src/analysis/special_functions/trigonometric.lean

```diff
@@ -2205,6 +2236,47 @@ begin
     ring },
 end
 
+lemma tan_add {x y : ℂ}
+  (h : ((∀ k : ℤ, x ≠ (2 * k + 1) * π / 2) ∧ ∀ l : ℤ, y ≠ (2 * l + 1) * π / 2)
+     ∨ ((∃ k : ℤ, x = (2 * k + 1) * π / 2) ∧ ∃ l : ℤ, y = (2 * l + 1) * π / 2)) :
```

Comment:
I'm not sure the second case will be very useful. Maybe you could keep this statement (or only the second case) as `tan_add'` and let `tan_add` be the statement assuming `(∀ k : ℤ, x ≠ (2 * k + 1) * π / 2) ∧ ∀ l : ℤ, y ≠ (2 * l + 1) * π / 2`.